controllers/asosecret_controller.go

```diff
@@ -206,19 +206,46 @@ func (asos *ASOSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	log = log.WithValues("cluster", cluster.Name)
 
-	// Return early if the AzureCluster or Cluster is paused.
+	// Return early if the ASO Secret Owner(AzureCluster or AzureManagedControlPlane) or Cluster is paused.
 	if annotations.IsPaused(cluster, asoSecretOwner) {
 		log.Info(fmt.Sprintf("%s or linked Cluster is marked as paused. Won't reconcile", asoSecretOwner.GetObjectKind()))
+		asos.Recorder.Eventf(asoSecretOwner, corev1.EventTypeNormal, "ClusterPaused",
+			fmt.Sprintf("%s or linked Cluster is marked as paused. Won't reconcile", asoSecretOwner.GetObjectKind().GroupVersionKind().Kind))
 		return ctrl.Result{}, nil
 	}
 
-	// Construct ASO secret for this AzureCluster
+	// Construct the ASO secret for this Cluster
+	newASOSecret, err := asos.createSecretFromClusterIdentity(ctx, clusterIdentity, cluster, azureClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	gvk := cluster.GetObjectKind().GroupVersionKind()
+	owner := metav1.OwnerReference{
```

Comment:
Would there be an issue keeping the AzureCluster/AzureManagedControlPlane as the owner of the Secret, but still name the Secret after the Cluster? I'm not sure it makes any practical difference whether the Secret's owner is the CAPZ resource or the Cluster, but having the AzureCluster/AzureManagedControlPlane as the owner seems slightly tidier to me.

Reply:
I'm fine either way. I put the Cluster object as the owner initially because the secret label key for owned is also set based on the Cluster name, e.g. :
```
Labels: map[string]string{
    cluster.GetName(): string(infrav1.ResourceLifecycleOwned),
},
```

I've adjusted the secret owner to point to the AzureCluster/AzureManagedControlPlane